src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -624,10 +624,20 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
-        case NI_Vector64_ConvertToInt32:
         case NI_Vector64_ConvertToInt32Native:
-        case NI_Vector128_ConvertToInt32:
         case NI_Vector128_ConvertToInt32Native:
+        {
+            if (opts.IsReadyToRun())
```

Comment:
Should we skip this block on platforms where `ConvertToInt32Native` is same as `ConvertToInt32` ?

Reply:
I had considered it and opted to not do that to avoid any potential future issues. The factor being that while they are the same today, there could be some future version of the platform where an alternative instruction is exposed that ends up being better but having different edge case behavior.

`ReciprocalSqrtEstimate` is one such example where `rsqrtss` was baseline in SSE but then `vrsqrt14ss` came out in AVX512/AVX10 and is faster and better handles various edge cases.

Such a scenario is less likely to happen for a cases `ConvertToInt32Native`, but its overall better to be consistent and the intrinsic expansion makes this much cheaper than a normal method (and is less likely to be used on a startup path where it might matter).